python/packages/autogen-ext/src/autogen_ext/agents/azure/_azure_ai_agent.py

```diff
@@ -768,18 +769,48 @@ async def on_messages_stream(
         if not agent_messages.data:
             raise ValueError("No messages received from assistant")
 
-        # Get the last message's content
-        last_message = agent_messages.data[0]
+        # Get the last message from the agent
+        last_message = agent_messages.get_last_message_by_role(models.MessageRole.AGENT)
+        if not last_message:
+            event_logger.debug("No message with AGENT role found, falling back to first message")
+            last_message = agent_messages.data[0]  # Fallback to first message
+
         if not last_message.content:
-            raise ValueError(f"No content in the last message: {last_message}")
+            raise ValueError(f"No content in the last message")
 
         # Extract text content
-        text_content = agent_messages.text_messages
-        if not text_content:
-            raise ValueError(f"Expected text content in the last message: {last_message.content}")
+        message_text = ""
+        for text_message in last_message.text_messages:
+            message_text += text_message.text.value
+
+        # Extract citations
+        citations: list[Any] = []
+
+        # Try accessing annotations directly
+        if hasattr(last_message, "annotations") and last_message.annotations:  # type: ignore
+            event_logger.debug(f"Found {len(last_message.annotations)} annotations")  # type: ignore
```

Comment:
Just noticed the use of `event_logger` is incorrect. We should be using `TRACE_LOGGER_NAME` and change the variable to `trace_logger`.